### PR TITLE
refactor TLB to chisel3.

### PR DESCRIPTION
**Pull request by SingularityKChen**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

Refactor TLB.scala to chisel3 style.

### CODE REVIEW

**Comment by sequencer** [comment]
TLB doc is merged, plz rebase to master.

**Comment by SingularityKChen** [comment]
> TLB doc is merged, plz rebase to master.

Rebased.

File: src/main/scala/rocket/TLB.scala [hunk 1]
```diff
@@ -307,18 +306,18 @@ case class TLBConfig(
...
+    val resp = Output(new TLBResp())
     /** SFence Input */
-    val sfence = Valid(new SFenceReq).asInput
+    val sfence = Input(Valid(new SFenceReq))
     /** IO to PTW */
     val ptw = new TLBPTWIO
     /** suppress a TLB refill, one cycle after a miss */
-    val kill = Bool(INPUT)
-  }
+    val kill = Input(Bool()) // suppress a TLB refill, one cycle after a miss
```
Comment (sequencer) [inline src/main/scala/rocket/TLB.scala:319]: why refactor the documentation code style?

**Review by sequencer — CHANGES_REQUESTED** [review]
Don't change anything other than chisel3 refactor.

File: src/main/scala/rocket/TLB.scala [hunk 2]
```diff
@@ -337,10 +336,10 @@ class TLB(instruction: Boolean, lgMaxSize: Int, cfg: TLBConfig)(implicit edge: T
   def all_entries = ordinary_entries ++ special_entry
   def all_real_entries = sectored_entries.flatten ++ superpage_entries ++ special_entry
 
-  val s_ready :: s_request :: s_wait :: s_wait_invalidate :: Nil = Enum(UInt(), 4)
-  val state = Reg(init=s_ready)
+  val s_ready :: s_request :: s_wait :: s_wait_invalidate :: Nil = Enum(4)
+  val state = RegInit(init=s_ready)
```
Comment (sequencer) [inline src/main/scala/rocket/TLB.scala:340]:
```suggestion
  val state = RegInit(s_ready)
```

**Review by sequencer — CHANGES_REQUESTED** [review]

File: src/main/scala/rocket/TLB.scala [hunk 3]
```diff
@@ -38,21 +37,21 @@ case object VMIdBits extends Field[Int](0)
...
-  val vaddr = UInt(width = vaddrBitsExtended)
+  val vaddr = UInt(vaddrBitsExtended.W)
   /** don't lookup TLB, bypass vaddr as paddr */
   val passthrough = Bool()
   /** granularity */
-  val size = UInt(width = log2Ceil(lgMaxSize + 1))
+  val size = UInt(log2Ceil(lgMaxSize + 1).W)
   /** memory command. */
-  val cmd  = Bits(width = M_SZ)
+  val cmd  = Bits(M_SZ.W)
```
Comment (sequencer) [inline src/main/scala/rocket/TLB.scala:54]:
```suggestion
  val cmd  = UInt(M_SZ.W)
```

File: src/main/scala/rocket/TLB.scala [hunk 4]
```diff
@@ -412,12 +411,12 @@ class TLB(instruction: Boolean, lgMaxSize: Int, cfg: TLBConfig)(implicit edge: T
...
   def fastCheck(member: TLManagerParameters => Boolean) =
-    legal_address && edge.manager.fastProperty(mpu_physaddr, member, (b:Boolean) => Bool(b))
+    legal_address && edge.manager.fastProperty(mpu_physaddr, member, (b:Boolean) => b.B)
   // todo: using DataScratchpad doesn't support cacheable.
-  val cacheable = fastCheck(_.supportsAcquireB) && (instruction || !usingDataScratchpad)
+  val cacheable = fastCheck(_.supportsAcquireB) && (instruction || !usingDataScratchpad).B
   val homogeneous = TLBPageLookup(edge.manager.managers, xLen, p(CacheBlockBytes), BigInt(1) << pgIdxBits)(mpu_physaddr).homogeneous
   // In M mode, if access DM address(debug module program buffer)
-  val deny_access_to_debug = mpu_priv <= PRV.M && p(DebugModuleKey).map(dmp => dmp.address.contains(mpu_physaddr)).getOrElse(false)
+  val deny_access_to_debug = mpu_priv <= PRV.M .U&& p(DebugModuleKey).map(dmp => dmp.address.contains(mpu_physaddr)).getOrElse(false.B)
```
Comment (sequencer) [inline src/main/scala/rocket/TLB.scala:419]:
```suggestion
  val deny_access_to_debug = mpu_priv <= PRV.M.U && p(DebugModuleKey).map(dmp => dmp.address.contains(mpu_physaddr)).getOrElse(false.B)
```

**Review by sequencer — CHANGES_REQUESTED** [review]

File: src/main/scala/rocket/TLB.scala [hunk 5]
```diff
@@ -38,21 +37,21 @@ case object VMIdBits extends Field[Int](0)
...
-  val vaddr = UInt(width = vaddrBitsExtended)
+  val vaddr = UInt(vaddrBitsExtended.W)
   /** don't lookup TLB, bypass vaddr as paddr */
   val passthrough = Bool()
   /** granularity */
-  val size = UInt(width = log2Ceil(lgMaxSize + 1))
+  val size = UInt(log2Ceil(lgMaxSize + 1).W)
   /** memory command. */
-  val cmd  = Bits(width = M_SZ)
+  val cmd  =UInt(M_SZ.W)
```
Comment (sequencer) [inline src/main/scala/rocket/TLB.scala:54]:
```suggestion
  val cmd  = UInt(M_SZ.W)
```

**Review by sequencer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**